### PR TITLE
deployment of Resource group, function app, storage account and application service plan 

### PR DESCRIPTION
**Pull request by jinudean**
completed [task] #80
1. Start research to Create and Deploy Resource group with ARM template on Microsoft documentation.
https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-tutorial-create-first-template?tabs=azure-powershell

2. Test blank Resource Group on Azure with North Seattle Colleges Subscription.
    a. after success with blank deployment
    b. add Storage account, function app and application service plan
    c. check required resource for function app follow MS document page
     https://docs.microsoft.com/en-us/azure/azure-functions/functions-infrastructure-as-code#app-service-plan
Screen Shot 2021-01-26 at 9 45 18 PM
   d. coding completed json template 
        i. Microsoft.Web/sites => a Function app
        ii. Microsoft.Storage/storageAccounts => Azure storage account
        iii. Microsoft.Web/serverfarms => App service Plan

3. Success create and deploy new resource group on North Seattle Colleges Azure
4. Verify deployment 
    a. sign in to the Azure portal 
    b. select named resource group
Screen Shot 2021-01-26 at 6 38 39 PM
   c. select Deployments at this time shows "1 Succeeded"
Screen Shot 2021-01-26 at 9 38 09 PM

   d. verify all 3 service is deployed on resource group.

### CODE REVIEW

**Comment by ahmadafrasiyab** [comment]
I have successfully run the script. There were no errors and a successful deployment was done as follows:

image

I did not see any hardcoding. The folder structure is reasonable as well. Since my Pull request number 54 has a schema error in template file, I suggest we close that one and build on top of this. Though there are some further improvements that can be done regarding validation and resource group name building but that is something that can be tackled in a future sprint.

File: Automation/task2/team2/deployment.ps1 [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    [Parameter(Mandatory=$true)]
+    [string]
+    $AppServiceplaneName,
+
+    [Parameter(Mandatory=$true)]
+    [string]
+    $DeployName
+)
+#open log
+$prefix = "jin-templa-deployment"
```
Comment (ahmadafrasiyab) [inline Automation/task2/team2/deployment.ps1:48]: This is pretty reasonable to add a logging function which can help with debugging. Thumbs up for this.  I am guessing we are overwriting the same log file again and again on each new deployment? or is the new input log appended to the previous log. But this is just a follow up question. All in all this looks good and I approve the merge.

**Review by ahmadafrasiyab — APPROVED** [review]
Runs good without errors. Please see my comments. I approve this.

File: Automation/task2/team2/deployment.ps1 (hunk 1 above)
Comment (yergu) [reply Automation/task2/team2/deployment.ps1:48]: I deployed it successfully and  created the required output , I didn't see any error, so I approve the merge

**Comment by yergu** [comment]
It runs successfully and created the 3 resources. I don't see error, I approved to be merge

**Review by yergu — APPROVED** [review]

File: Automation/task2/team2/deployment.ps1 [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+$stamp = (Get-Date).toString().Replace("/","-").Replace(":","-")
+Start-Transcript "./$prefix-$stamp.log"
+
+$credentials = New-Object -TypeName System.Management.Automation.PSCredential($ServicePrincipalId, $ServicePrincipalPassword );
+
+Connect-AzAccount -Credential $credentials -ServicePrincipal -Tenant $TenantId -Subscription $SubscriptionId;
+
+Set-AzContext $SubscriptionId
```
Comment (toddysm) [inline Automation/task2/team2/deployment.ps1:56]: You already have this one in `Connect-AzAccount` - it is not needed here.

**Pull request merged**